lib/core/versions/latest/models/DocumentModel.ts

```diff
@@ -6,6 +6,6 @@ import ServiceEndpointModel from './ServiceEndpointModel';
  * NOTE: This model should ONLY be used by the `DocumentComposer`.
  */
 export default interface DocumentModel {
-  public_keys?: PublicKeyModel[];
-  service_endpoints?: ServiceEndpointModel[];
+  publicKey?: PublicKeyModel[];
```

Comment:
This is an internal model so they don't have to have the same key names as DID spec, and we want array names to be plurals.

Reply:
since these keys expressed in the external (client facing) API, I believe the choice to add and remove `s` will lead to significant developer pain... and since did core uses `publicKey` I think the external interface should match the external representation. If these data models were not externally facing I would happily add an `s` to resolve this.

Reply:
I disagree with @OR13 here. But since this is contentious and technically unrelated to the casing change, please stash this part of the change for a separate discussion.

Reply:
Preference is towards plural here to adhere to the style guide.